### PR TITLE
Handle "429 too many requests exception" in S3 archiving

### PR DESCRIPTION
**Pull request by patrickmann**
Resolves Graylog2/graylog-plugin-enterprise#5136

See issue for details

### CODE REVIEW

File: graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        };
+        when(response.getEntity()).thenReturn(httpEntity);
+
+        ResponseException responseException = new ResponseException(response);
+        RestStatus restStatus = RestStatus.BAD_REQUEST;
+        OpenSearchStatusException statusException = new OpenSearchStatusException(
+                "status msg", restStatus, responseException);
+
+        OpenSearchInstance openSearchInstance = OpenSearchInstance.create();
+        final OpenSearchClient openSearchClient = openSearchInstance.openSearchClient();
```
Comment (mpfz0r) [inline graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java:143]:
```suggestion
        final OpenSearchClient openSearchClient = new OpenSearchClient(mock(RestHighLevelClient.class), true, new ObjectMapper());```
        
If we are going to throw the exception ourselves anyway, we don't need a real opensearch instance, which takes minutes to start, but can also just mock the client.

Comment (mpfz0r) [reply graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java:143]: also we should've called
`openSearchInstance.close()` to shut down the test instance properly

**Review by mpfz0r — CHANGES_REQUESTED** [review]
test can be made faster

File: graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java (hunk 1 above)
Comment (patrickmann) [reply graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/OpenSearchExceptionTest.java]: Good point. Removed the instance.

**Pull request merged**